pkg/apis/psmdb/v1/psmdb_defaults.go

```diff
@@ -438,6 +438,10 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
 		return errors.New("backup.enabled must be false on unmanaged clusters")
 	}
 
+	if cr.Spec.Unmanaged && cr.Spec.UpdateStrategy == SmartUpdateStatefulSetStrategyType {
```

Comment:
should we set `cr.Spec.ClusterServiceDNSSuffix` before return from the function?

Reply:
fixed